tests/deepspeed/test_model_zoo.py

```diff
@@ -0,0 +1,259 @@
+    slow,
+)
+from transformers.trainer_utils import set_seed
+
+
+tests_dir = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
+root_dir = os.path.dirname(tests_dir)
+with ExtendSysPath(tests_dir):
+    from test_trainer import TrainerIntegrationCommon  # noqa
+
+    if is_torch_available():
+        from test_trainer import RegressionModelConfig, RegressionPreTrainedModel, get_regression_trainer  # noqa
+
+
+set_seed(42)
+
+# translation
+FSMT_TINY = "stas/tiny-wmt19-en-de"
+BART_TINY = "sshleifer/bart-tiny-random"
+T5_SMALL = "t5-small"
+T5_TINY = "patrickvonplaten/t5-tiny-random"
+MBART_TINY = "sshleifer/tiny-mbart"
+MARIAN_TINY = "sshleifer/tiny-marian-en-de"
+
+# summarization
+PEGASUS_TINY = "stas/pegasus-cnn_dailymail-tiny-random"
+
+# causal lm
+GPT2_TINY = "sshleifer/tiny-gpt2"
+
+# question-answering
+DISTILBERT_TINY = "sshleifer/tiny-distilbert-base-cased-distilled-squad"
+ROBERTA_TINY = "sshleifer/tiny-distilroberta-base"
+XLM_ROBERTA_TINY = "hf-internal-testing/tiny-xlm-roberta"
+
+# masked lm
+ELECTRA_TINY = "hf-internal-testing/tiny-electra"
+
+# classification
+XLNET_TINY = "sshleifer/tiny-xlnet-base-cased"
+BERT_TINY = "hf-internal-testing/tiny-bert"
+
+
+# TODO: to add:
+# albert
+# deberta
+# funnel
+# longformer
+# dpr
+# gpt_neo
+# camembert
+# deberta-v2
+# m2m_100
+# tapas
+# vit
+# big_bird
+
+
+def get_launcher(distributed=False):
+    # 1. explicitly set --num_nodes=1 just in case these tests end up run on a multi-node setup
+    # - it won't be able to handle that
+    # 2. for now testing with just 2 gpus max (since some quality tests may give different
+    # results with mode gpus because we use very little data)
+    num_gpus = min(2, get_gpu_count()) if distributed else 1
+    return f"deepspeed --num_nodes 1 --num_gpus {num_gpus}".split()
+
+
+def make_task_cmds():
+    data_dir_fixtures = f"{tests_dir}/fixtures"
+    data_dir_samples = f"{data_dir_fixtures}/tests_samples"
+    data_dir_wmt = f"{data_dir_samples}/wmt_en_ro"
+    data_dir_xsum = f"{data_dir_samples}/xsum"
+    args_main = """
+        --do_train
+        --max_train_samples 4
+        --per_device_train_batch_size 2
+        --num_train_epochs 1
+        --fp16
+        --report_to none
+        --overwrite_output_dir
+        """.split()
+
+    # XXX: try to cover as many models as possible once (it's enough to run on one task per model)
+    # but need a tiny model for each
+    #
+    # should have T5_TINY, etc. global var defined
+    tasks2models = dict(
+        trans=[
+            "bart",
+            "fsmt",
+            "marian",
+            "mbart",
+            "t5",
+        ],
+        sum=[
+            "pegasus",
+        ],
+        clm=[
+            "gpt2",
+            "xlm-roberta",
+        ],
+        mlm=[
+            "electra",
+        ],
+        qa=[
+            "distilbert",
+            "roberta",
+        ],
+        clas=[
+            "bert",
+            "xlnet",
+        ],
+    )
```

Comment:
Most of the lists fit on one line, it would be easier to read this way.

Reply:
I'm pretty sure it was auto-formatter that did the formatting, but I actually need it this way at the moment as I'm testing multiple models and so want to easily comment out specific lines. I have a lot more models to add and not all go easily. So it will be a bit of a process. When done I'm happy to reformat this.